### PR TITLE
Upgrade dependencies to match quality checklist

### LINKED ISSUE #46: Code Exchange quality checklist
The purpose of this checklist is to assess the state of this repository, and bring it to an acceptable level for long-term support by completing the tasks listed below.

> Please check every part of this checklist for us to process your request. You can do this by marking them with [x] instead of [ ].

### Documentation

- [x] Twilio [LICENSE](https://github.com/twilio-labs/code-exchange/blob/master/LICENSE) exists in repository

- [x] [Code of Conduct](https://github.com/twilio-labs/.github/blob/master/CODE_OF_CONDUCT.md) exists in repository

- [x] [CONTRIBUTING](https://github.com/twilio-labs/.github/blob/master/CONTRIBUTING.md) exists in repository 

- [x] No broken links in README.md

- [x] Build status exists at top of README.md

- [x] Set up steps are complete and accurate

- [x] CodeExchange repository is linked in README.md

### GitHub

- [x] Dependabot set up

- [x] Dependabot automerge set up

- [x] Uses GitHub Actions (required for Twilio sample) or other CI

### Code

- [x] Packages/dependencies are up to date

- [x] Code compiles against latest packages/dependencies

- [x] Matches [CodeExchange Guidelines](https://github.com/twilio-labs/code-exchange/tree/master/guidelines) to the best of ability

### Tests

- [x] Functionality tests have been created

- [x] All tests pass

### App

- [x] Confirmed sample behaves as expected (after dependencies have been updated)

- [x] This file has been copied into a wiki entry for the respective repository or added as a GitHub issue

### General

- [x] I've read and followed the contribution guidelines

- [x] I've read and implemented my template according to the respective coding guidelines for my project

- [x] I agree and adhere to the Code of Conduct

- [x] The template has been authored by me / or my organization and will be kept up-to-date. If the template will not longer be maintained, I'll notify this project.

### LINKED ISSUE #48: Create steps in markdown to create Account Security API Key
Include steps to go to Authy section of Console to create Account Security API Key.

### PR DESCRIPTION
**Pull request by po5i**
Resolves #46

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches: [ master, next ]
+  pull_request:
+    branches: [ master, next ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (mmena1) [inline .github/workflows/ruby.yml]: This should also be able to run on mac and linux. Please refer to this [example](https://github.com/TwilioDevEd/sample-template-rails/blob/master/.github/workflows/rails.yml)

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup System
+      run: |
+        sudo apt-get install libsqlite3-dev
+    - name: Set up Ruby 2.6
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
```
Comment (mmena1) [inline .github/workflows/ruby.yml]: The newest ruby version is `2.7.1` and we should upgrade to that version. On this [ruby template](https://github.com/TwilioDevEd/sample-template-rails) we use version `2.6.x` because at that time, `2.7.0` had some issues with the Windows CI

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+install:
+	bundle install
+
+database:
+	bundle exec rails db:migrate
+
+serve:
+	bundle exec rails server --binding 0.0.0.0
```
Comment (mmena1) [inline Makefile:10]: As a good practice, all Makefiles should include a `.PHONY` [entry](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html).

**TL;DR**
By default, the `make` command targets files, so `make install` is actually looking for a file named `install` but since it doesn't exist it will execute the rule `install` on the Makefile. By adding `.PHONY: install`, we are telling the `make` command that `install` is not a file and should always run as the name of the recipe to be executed, even if there is a file named `install`. So, as a rule of thumb, all rules defined on the Makefile should be declared as `.PHONY` targets to avoid any potential collisions with similar named files.

File: README.md [hunk 4]
```diff
@@ -1,18 +1,27 @@
 ![Twilio Logo](./twilio_logo_red.png)
 # Twilio Account Security Quickstart - Twilio Authy and Twilio Verify
+
+![](https://github.com/TwilioDevEd/account-security-quickstart-rails/workflows/Ruby/badge.svg)
+
 > We are currently in the process of updating this sample template. If you are encountering any issues with the sample, please open an issue at [github.com/twilio-labs/code-exchange/issues](https://github.com/twilio-labs/code-exchange/issues) and we'll try to help you.
```
Comment (mmena1) [inline README.md:7]: This should be removed. Talk to Dominik to know if this isn't needed anymore

File: app/controllers/api/verify_controller.rb [hunk 5]
```diff
@@ -1,48 +1,68 @@
...
+  def set_twilio_client
+    @client = Twilio::REST::Client.new(ENV['TWILIO_ACCOUNT_SID'], ENV['TWILIO_AUTH_TOKEN'])
+  end
+
+  def start_verification(to, channel = 'sms')
+    channel = 'sms' unless ['sms', 'voice'].include? channel
+    verification = @client.verify.services(ENV['VERIFY_SERVICE_SID'])
+                          .verifications
+                          .create(:to => to, :channel => channel)
+    verification
```
Comment (mmena1) [inline app/controllers/api/verify_controller.rb]: We don't need this line. Ruby always returns the evaluated expression on the last line of a function unless there is an explicit `return` statement after it.

File: app/controllers/api/verify_controller.rb [hunk 6]
```diff
@@ -1,48 +1,68 @@
...
+  end
+
+  def check_verification(phone, code)
+    verification_check = @client.verify.services(ENV['VERIFY_SERVICE_SID'])
+                                .verification_checks
+                                .create(:to => phone, :code => code)
+    verification_check
```
Comment (mmena1) [inline app/controllers/api/verify_controller.rb]: Here, same as above

File: spec/controllers/twofa_controller_spec.rb [hunk 7]
```diff
@@ -1,5 +1,152 @@
...
+          allow(Authy::API).to receive(:request_sms).and_return(APIRespose.new(false))
+          post :sms, session: { username: 'angie' }
+
+          expect(response.body).to eql("{\"err\":\"Error requesting SMS token\"}")
+          expect(response.status).to eq(500)
+        end
+      end
+    end
+
+    context 'when a user not exists ' do
```
Comment (mmena1) [inline spec/controllers/twofa_controller_spec.rb]:
```suggestion
    context 'when a user does not exist ' do
```

File: spec/controllers/twofa_controller_spec.rb [hunk 8]
```diff
@@ -1,5 +1,152 @@
...
+          allow(Authy::API).to receive(:request_phone_call).and_return(APIRespose.new(false))
+          post :voice, session: { username: 'angie' }
+
+          expect(response.body).to eql("{\"err\":\"Error requesting Phone call token\"}")
+          expect(response.status).to eq(500)
+        end
+      end
+    end
+
+    context 'when a user not exists ' do
```
Comment (mmena1) [inline spec/controllers/twofa_controller_spec.rb]:
```suggestion
    context 'when a user does not exist ' do
```

File: spec/controllers/twofa_controller_spec.rb [hunk 9]
```diff
@@ -1,5 +1,152 @@
...
+          allow(Authy::API).to receive(:verify).and_return(APIRespose.new(false))
+          post :verify, session: { username: 'angie' }
+
+          expect(response.body).to eql("{\"err\":\"Verify Token Error\"}")
+          expect(response.status).to eq(500)
+        end
+      end
+    end
+
+    context 'when a user not exists ' do
```
Comment (mmena1) [inline spec/controllers/twofa_controller_spec.rb]:
```suggestion
    context 'when a user does not exist ' do
```

File: spec/controllers/twofa_controller_spec.rb [hunk 10]
```diff
@@ -1,5 +1,152 @@
...
+          allow(Authy::OneTouch).to receive(:send_approval_request).and_return(APIRespose.new(false))
+          post :onetouch, session: { username: 'angie' }
+
+          expect(response.body).to eql("{\"err\":\"Create Push Error\"}")
+          expect(response.status).to eq(500)
+        end
+      end
+    end
+
+    context 'when a user not exists ' do
```
Comment (mmena1) [inline spec/controllers/twofa_controller_spec.rb]:
```suggestion
    context 'when a user does not exist ' do
```

**Review by mmena1 — CHANGES_REQUESTED** [review]
Nice work Angie! I see a lot of improvements over the old code. I left some minor comments about grammar and code cleanup. I do have a concern though:

Rails 6 brought some [big changes](https://edgeguides.rubyonrails.org/6_0_release_notes.html#railties-notable-changes) over its predecessor, one of them is the inclusion of `webpack` to serve front-end assets which is now the default JavaScript compiler. In order to get the new Rails 6 structure we need to add a NodeJS engine to provide the webpack feature which will be consumed through [Webpacker](https://github.com/rails/webpacker). The easiest way to add this is to create a new project from scratch (or use the [sample template](https://github.com/TwilioDevEd/sample-template-rails)) then copy all the content to the new project and move the contents from the public directory to the new assets path.

_Other minor stuff:_

- This [image](https://github.com/TwilioDevEd/account-security-quickstart-rails/blob/783a175c6bda973be95579b7cee807852e7d7430/api_key.png) should be removed.

**Comment by rojasTob** [comment]
> Nice work Angie! I see a lot of improvements over the old code. I left some minor comments about grammar and code cleanup. I do have a concern though:
> 
> Rails 6 brought some [big changes](https://edgeguides.rubyonrails.org/6_0_release_notes.html#railties-notable-changes) over its predecessor, one of them is the inclusion of `webpack` to serve front-end assets which is now the default JavaScript compiler. In order to get the new Rails 6 structure we need to add a NodeJS engine to provide the webpack feature which will be consumed through [Webpacker](https://github.com/rails/webpacker). The easiest way to add this is to create a new project from scratch (or use the [sample template](https://github.com/TwilioDevEd/sample-template-rails)) then copy all the content to the new project and move the contents from the public directory to the new assets path.
> 
> _Other minor stuff:_
> 
> * This [image](https://github.com/TwilioDevEd/account-security-quickstart-rails/blob/783a175c6bda973be95579b7cee807852e7d7430/api_key.png) should be removed.

Hey @mrcrow85 , about your comment of the NodeJS engine, thanks for pointing this out. I'm going to talk about this with Dominik and check if it is possible to do this in this phase.

**Review by mmena1 — COMMENTED** [review]
Looks good 👍

**Review by mmena1 — APPROVED** [review]
